src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -383,6 +383,46 @@ public StringSegment Trim()
         }
 
         /// <summary>
+        /// Removes all leading whitespaces.
+        /// </summary>
+        /// <returns>The trimmed <see cref="StringSegment"/>.</returns>
+        public StringSegment TrimStart()
```

Comment:
It's a struct, so it doesn't have any allocation costs. There might be costs to passing the struct around, but let's not prematurely optimize that.